fusillade/clouddirectory.py

```diff
@@ -999,91 +991,151 @@ def _add_links_batch(self, links: List[str], link_type: str):
+    def _remove_typed_links_batch(self, links: List[str], object_type, link_type: str, attributes: Dict,
+                                  incoming=False):
         """
         Removes links from this object in CloudDirectory.
         """
         if not links:
             return []
-        parent_path = self.cd.get_obj_type_path(link_type)
+        if incoming:
+            source, target = None, self.object_ref
+        else:
+            source, target = self.object_ref, None
+        link_path = self.cd.get_obj_type_path(object_type)
         batch_detach_typed_link = self.cd.batch_detach_typed_link
         make_typed_link_specifier = self.cd.make_typed_link_specifier
         operations = []
         for link in links:
-            parent_ref = f"{parent_path}{self.hash_name(link)}"
+            if source:
+                target = f"{link_path}{self.hash_name(link)}"
+            else:
+                source = f"{link_path}{self.hash_name(link)}"
             typed_link_specifier = make_typed_link_specifier(
-                parent_ref,
-                self.object_ref,
-                'association',
-                {'parent_type': link_type, 'child_type': self.object_type}
+                source,
+                target,
+                link_type,
+                attributes
             )
             operations.append(batch_detach_typed_link(typed_link_specifier))
         return operations
 
-    def lookup_policies(self) -> List[str]:
-        policy_paths = self.cd.lookup_policy(self.object_ref)
-        return self.cd.get_policies(policy_paths)
-
     @property
     def name(self):
         if not self._name:
             self._get_attributes(self._attributes)
             self._path_name = self.hash_name(self._name)
         return self._name
 
-    def create_policy(self, statement: str, policy_type='IAMPolicy') -> str:
+    def _get_attributes(self, attributes: List[str]):
+        """
+        retrieve attributes for this from CloudDirectory and sets local private variables.
+        """
+        if not attributes:
+            attributes = self._attributes
+        try:
+            resp = self.cd.get_object_attributes(self.object_ref, self._facet, attributes)
+        except cd_client.exceptions.ResourceNotFoundException:
+            raise FusilladeHTTPException(status=404, title="Not Found", detail="Resource does not exist.")
+        for attr in resp['Attributes']:
+            self.__setattr__('_' + attr['Key']['Name'], attr['Value'].popitem()[1])
+
+    def get_attributes(self, attributes: List[str]) -> Dict[str, str]:
+        """
+        retrieve the attribute values.
+        :param attributes:
+        :return:
+        """
+        try:
+            resp = self.cd.get_object_attributes(self.object_ref, self._facet, attributes)
+        except cd_client.exceptions.ResourceNotFoundException:
+            raise FusilladeHTTPException(status=404, title="Not Found", detail="Resource does not exist.")
```

Comment:
Nit: Consider using `requests.codes.not_found` instead of `404`.  >.>

Reply:
added error FusilladeNotFoundException